### DISCUSSION TITLE
通过rclone挂载webdav为本地硬盘，阿里网盘的文件不能播放

### DISCUSSION THREAD

**Discussion by wahaha3a** [AlistGo/alist#402] [Q&A]
阿里网盘的文件不能播放，天翼网盘和pikpak可以正常播放

**Comment by wahaha3a** [comment]
我是用群晖挂载的，同样的方法用aliyundrive-webdav挂载也可以正常播放

**Reply by wdjoys** [reply]
请问解决了吗，我和你的问题一样，通过alist的webdav，使用rclone挂载到nas，天翼云和OneDrive的视频播放正常，阿里云盘的能看到列表不能播放，阿里云盘的文件下载到本地后也不能正常播放。

**Comment by xhofe** [comment]
是否使用移动端token

**Reply by wahaha3a** [reply]
刚才试过了 还是不行，aliyundrive-webdav本身要求一定要网页端的token，换了之后媒体文件还是播放不了，用nplayer和plex都可以看到文件，但是点播放就提示错误

**Reply by xhofe** [reply]
本程序需要使用移动端token

**Reply by wahaha3a** [reply]
两种都试过 尴尬，还是专门装个es文件管理，到阿里网盘app里面提取的

**Reply by xhofe** [reply]
发一下 你的链接和webdav账密：support@nn.ci

**Reply by wahaha3a** [reply]
感谢回复，如果只是我自己的问题那就不必这么麻烦了哈，谢谢

**Comment by jsdfhasuh** [comment]
请问这个问题解决了吗

**Reply by wahaha3a** [reply]
没，我试过好几种方法，群晖挂载、openwrt挂载，用sharelist的，就阿里不行，最后折中dokcer挂了个webdav-aliyundriver，这个完美支持rclone

**Comment by jsdfhasuh** [comment]
我的也挂载不上，用raidrive能挂载上，但rclone挂载不上。我添加了天翼云盘和阿里网盘两个网盘

**Comment by jsdfhasuh** [comment]
rclone mount -v alist /volume1/阿里云/alist --allow-other --allow-non-empty --buffer-size 32M --vfs-read-chunk-size=32M --vfs-read-chunk-size-limit 2048M --vfs-cache-mode writes --log-file=/volume2/web/云盘电视剧.txt --daemon

**Comment by hcwhan** [comment]
一样 通过 rclone 挂载 阿里云盘  rclone 一直报错 ERROR : IO error: 403 Forbidden
raidrive 正常

**Comment by wdjoys** [comment]
请问解决了吗，我和你的问题一样，通过alist的webdav，使用rclone挂载到nas，天翼云和OneDrive的视频播放正常，阿里云盘的能看到列表不能播放，阿里云盘的文件下载到本地后也不能正常播放。


@Xhofe 

大大你好，
我通过这个方法获取的阿里云盘的token
https://github.com/Xhofe/alist/issues/88#issuecomment-907012079
alist的web端视频播放正常，通过alist提供的webdav，使用rclone挂载到本地，能看到列表，视频不能播放。

**Comment by xhofe** [comment]
我没有使用过rclone，可能是rclone添加了特定的referer头，可以尝试在添加账号时打开webdav代理通过alist中转，是否可以解决问题。

**Reply by wdjoys** [reply]
感谢大大回复，新年快乐🎇🎇🎇。
开启代理也不能解决，暂时放弃了。

**Reply by chrisdong** [reply]
rclone mount alist:/ /volume1/Mount/Rclone/alist --cache-dir /tmp --vfs-cache-mode writes --allow-non-empty --allow-other --file-perms 0777 --header "Referer:https://www.aliyundrive.com/drive"

我加上了 --header "Referer:https://www.aliyundrive.com/drive" 就可以了

**Comment by gzxiexl** [comment]
希望大佬关注下 rclone通过webdav挂载的功能不太稳定 阿里云盘时不时无法连接 且报错
Error #01: write tcp 172.18.0.8:5244->10.253.0.3:4510: write: broken pipe
2022/02/21 03:57:57 write tcp 172.18.0.8:5244->10.253.0.3:4499: write: broken pipe
GET /p/%E9%98%BF%E9%87%8C%E4%BA%91%E7%9B%98/TV/%E7%A0%B4%E4%BA%A7%E5%A7%90%E5%A6%B9%20(2011-2016)/Season%201/%E7%A0%B4%E4%BA%A7%E5%A7%90%E5%A6%B9.2.Broke.Girls%20(2011).H265.1080P.S01E19.mkv?sign=2c2265e747479d38 HTTP/1.1
Host: 10.0.0.254:5244
Accept: */*
Accept-Encoding: identity;q=1, *;q=0
Accept-Language: zh-CN,zh;q=0.9
Connection: keep-alive

trilium-device=desktop; trilium.sid=s%3A4kkWEqheMD_0SFSe9vGbHcPr-YS_QVbD.FUsBQQuRCqmnv8jRleMHR2v%2BGVGI%2Fx9j3lUMdoKF2nk; docker_listview_mode=advanced; ca_startupButton=onlynew; db-box1=0%3B1%3B2%3B3%3B4%3B5%3B6%3B8%3B953; diskio=diskio; organizr_user_uuid=c62c4541-3170-43b1-8082-20007e730924; organizr_token_cf10973d-f9a8-4eb6-9b08-0de9c7b741a6=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiIsImp0aSI6IjRmMWcyM2ExMmFhIn0.eyJpc3MiOiJPcmdhbml6ciIsImF1ZCI6Ik9yZ2FuaXpyIiwianRpIjoiNGYxZzIzYTEyYWEiLCJpYXQiOjE2NDQ5NzcxMDUsImV4cCI6MTY0NTU4MTkwNSwibmFtZSI6Imd6eGlleGwiLCJncm91cCI6IkFkbWluIiwiZ3JvdXBJRCI6MCwiZW1haWwiOiJ4Z3h4bDg5QGdtYWlsLmNvbSIsImltYWdlIjoiaHR0cHM6XC9cL3d3dy5ncmF2YXRhci5jb21cL2F2YXRhclwvMTJlZjEwYzI5NDQzYWI5YmM3ODQ0NTY1NzdlZDMxMDU_cz0xMDAmZD1tbSIsInVzZXJJRCI6MX0.2Zq8KbUgtpOQy4u_Ru-mrxg-xrAIeCIGCksZJfNMv3Q; organizrLanguage=zh; default-theme=ngax; unraid_45c868fd8f15882827387903d4e6ffa3=6a3dc5752887a421656432d47a17f68d; ud_reload=UD_reload; UPC_ENV=local; auth=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7ImlkIjoxLCJsb2NhbGUiOiJ6aC1jbiIsInZpZXdNb2RlIjoibGlzdCIsInNpbmdsZUNsaWNrIjpmYWxzZSwicGVybSI6eyJhZG1pbiI6dHJ1ZSwiZXhlY3V0ZSI6dHJ1ZSwiY3JlYXRlIjp0cnVlLCJyZW5hbWUiOnRydWUsIm1vZGlmeSI6dHJ1ZSwiZGVsZXRlIjp0cnVlLCJzaGFyZSI6dHJ1ZSwiZG93bmxvYWQiOnRydWV9LCJjb21tYW5kcyI6W10sImxvY2tQYXNzd29yZCI6ZmFsc2UsImhpZGVEb3RmaWxlcyI6ZmFsc2UsImRhdGVGb3JtYXQiOnRydWV9LCJleHAiOjE2NDU0MjIwMTcsImlhdCI6MTY0NTQxNDgxNywiaXNzIjoiRmlsZSBCcm93c2VyIn0.NYx1Aucwp3rYCx2qamV9IuOs-AuwHThtEphE5LW0-Hg; cpu-init=1%3B1%3B1%3B3%3B3%3B1%3B1%3B4%3B2%3B3%3B1%3B1%3B1%3B1%3B5%3B1%3B1%3B2%3B1%3B3%3B2%3B1%3B1%3B1%3B2%3B3%3B3%3B1%3B1%3B3; rxd-init=95580.184686503%3B112914.55741206%3B225045.60360936%3B252015.61130629%3B75707.779814541%3B156449.67553477%3B540207.05296785%3B224292.45297981%3B158360.46175013%3B106530.39650542%3B100722.65826416%3B208358.53077475%3B240839.75019372%3B307971.37162862%3B259597.61090995%3B127882.95205908%3B419048.13183326%3B127297.47728228%3B76511.83036438%3B73670.215909945%3B138506.38962759%3B160784.65662118%3B172005.93424516%3B151616.61683012%3B131939.78693986%3B79954.263628314%3B102735.37038775%3B102715.44164636%3B74272.818584381%3B119329.36285335; txd-init=5157147.2424235%3B8937198.3956619%3B5920158.4916296%3B34101683.002034%3B663496.78773146%3B1657976.9931532%3B75488459.502027%3B25806997.185986%3B1500700.2773655%3B1267090.5173316%3B2963147.7252257%3B16815139.280072%3B12659146.853639%3B14979710.594074%3B14900122.20556%3B967515.43200355%3B87973262.102217%3B718523.89321707%3B680170.77862049%3B611102.70462028%3B10114159.121651%3B13548176.695414%3B13563803.020291%3B15043515.029032%3B1936859.7873567%3B624179.69065746%3B150588.21474834%3B90710.155336438%3B502189.61646817%3B1817330.6496831
Range: bytes=0-
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36

建议docker能否添加本地挂载功能 直接绕过rclone挂载到本地最好

**Reply by gzxiexl** [reply]
夸克网盘可以正常播放 但是日志看也会出现以上报错

**Reply by gzxiexl** [reply]
使用ios的webdav客户端播放夸克网盘的文件没问题 通过rclone挂载到本地则夸克里无法播放视频 阿里云盘的则可以 但是反应速度有卡顿

**Reply by xhofe** [reply]
write tcp 是正常错误，因为客户端关闭了链接，输出流无法再写入，直接用的io.Copy就会抱着个错，可以不用管。

**Reply by gzxiexl** [reply]
我是通过unraid平台docker安装的alist，添加了夸克、阿里云盘等账号，web端非常ok，速度非常满意，手机端通过app以webdav协议添加alist，列表和播放都非常ok，速度跑满，但是就是rclone通过webdav挂载alist到本地就不正常了，阿里还可以，夸克网盘只有列表，不能播放视频。查看日志，发现大量报错。截取一小段如下：
[GIN] 2022/02/21 - 08:19:18 | 200 | 100.741348ms | 10.0.0.254 | GET "/p/夸克网盘/影视/金鱼妻/[SHANA]金鱼妻.S01E08.WEBRip.Netflix.zh-Hans.mp4"
ERRO[2022-02-21 08:19:18] <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN"><html><head><title>403 Forbidden</title></head><body><h1>403 Forbidden</h1><p>You don't have permission to access the URL on this server.</p><p>denied by Referer ACL</p><hr/>Powered by Tengine <br/>CDN Request Id: db906c9816454315584436959e</body></html>

**Reply by gzxiexl** [reply]
如果在ios app上通过webdav播放夸克网盘里视频 则正常 速度跑满
[GIN] 2022/02/21 - 08:21:39 | 200 | 1.130336948s | 10.253.0.2 | GET "/p/夸克网盘/影视/金鱼妻/[SHANA]金鱼妻.S01E01.WEBRip.Netflix.zh-Hans.mp4"
Error #01: write tcp 172.18.0.8:5244->10.253.0.2:56864: write: broken pipe

ERRO[2022-02-21 08:21:43] write tcp 172.18.0.8:5244->10.253.0.2:60113: write: broken pipe
[GIN] 2022/02/21 - 08:21:43 | 200 | 5.588259312s | 10.253.0.2 | GET "/p/夸克网盘/影视/金鱼妻/[SHANA]金鱼妻.S01E01.WEBRip.Netflix.zh-Hans.mp4"
Error #01: write tcp 172.18.0.8:5244->10.253.0.2:60113: write: broken pipe



2022/02/21 08:21:43 write tcp 172.18.0.8:5244->10.253.0.2:60113: write: broken pipe
GET /p/%e5%a4%b8%e5%85%8b%e7%bd%91%e7%9b%98/%e5%bd%b1%e8%a7%86/%e9%87%91%e9%b1%bc%e5%a6%bb/%5BSHANA%5D%e9%87%91%e9%b1%bc%e5%a6%bb.S01E01.WEBRip.Netflix.zh-Hans.mp4 HTTP/1.1
Host: 10.0.0.254:5244
Accept: */*
Accept-Encoding: identity
Accept-Language: zh-CN,zh-Hans;q=0.9
Connection: keep-alive
Range: bytes=0-1167130623
User-Agent: AppleCoreMedia/1.0.0.19D52 (iPhone; U; CPU OS 15_3_1 like Mac OS X; zh_cn)
X-Playback-Session-Id: D5D66ACA-099C-4E17-85C1-0AB53823757F


ERRO[2022-02-21 08:22:37] stream error: stream ID 11; PROTOCOL_ERROR; received from peer

[GIN] 2022/02/21 - 08:22:37 | 200 | 58.715692876s | 10.253.0.2 | GET "/p/夸克网盘/影视/金鱼妻/[SHANA]金鱼妻.S01E01.WEBRip.Netflix.zh-Hans.mp4"
